### ISSUE TITLE
Round-trip literals faithfully

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/syn#125]
```rust
extern crate syn;

#[macro_use]
extern crate quote;

fn main() {
    let ref int = syn::parse_token_trees("0_0").unwrap()[0];
    println!("{}", quote!(#int));
}
```

```
0
```

This should print 0_0. Relevant for making sure not to corrupt input to other macros: #124.

**Comment by mystor** [comment]
This should be fixed by #147, as we faithfully parse integers into `Literal` objects and don't interpret them now.

**Comment by mystor** [comment]
This is fixed on master 🎊 (see #158)
